spec/test_collector/test_links/reporter_spec.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'buildkite/test_collector/test_links_plugin/reporter'
+
+RSpec.describe Buildkite::TestCollector::TestLinksPlugin::Reporter do
+  let(:examples) { RSpec.world.filtered_examples }
+  let(:http_client) { double('Buildkite::TestCollector::HTTPClient') }
+  let(:suite_url) { 'https://example.com/suite/12345' }
+  let(:response) { { suite_url: suite_url }.to_json }
+  let(:io) { StringIO.new }
+  let(:reporter) { Buildkite::TestCollector::TestLinksPlugin::Reporter.new(io) }
+
+  before do
+    allow(Buildkite::TestCollector::HTTPClient).to receive(:new).and_return(http_client)
+    Buildkite::TestCollector.configure(
```

Comment:
💅 just a small style thing - would be good to have a line break here just for clarity